.github/CONTRIBUTING.md

```diff
@@ -219,6 +238,54 @@ Vignettes can be found in the `vignettes/` folder as .Rmd files. To start a new
+
+We have a `get_ons_api_data()` function that acts as a wrapper to the ONS API, it does things like converting readable parameters into a query string and also handles batching and multiple requests if needed, so you get all of the data in one nice neat data frame (there's a limit on the rows per single query for the API).
+
+The way ONS publish has varied over their first few years of publishing, and on top of that each data set has an individual API connection for every year of boundaries. As there's no link over time from the ONS side we have helper functions defined in R/datasets_utils.R that wrap these up into a single neat time series bundle for us. Given the likelihood of further variations, don't be too surprised if adding new years to the data sets results in errors first time around, some manual fudgery is often needed so roll up your sleeves and prepare to get elbow deep into the murky depths of the R/datasets_utils.R file!
+
+There is also some data we just define ourselves in code as we curate that, like custom regions we publish in DfE or our own lookup table for the shorthands used in the column names by ONS.
+
+#### Workflow for updating geography data
+
+Our general workflow for data in the package is:
+
+1. Add a new year into the relevant script in data-raw/ script
+2. Run the script to create a new data set
+3. Run all package checks to make sure the data hasn't gone all funky on you
+4. Update any fetch_ functions that use the data if appropriate
+5. Document any changes to the data set in R/datasets_documentation.R if appropriate
+
+Most data sets have tests that will fail as soon as the number of rows or columns change, this is both to provide a reliable service to users, but also to catch and remind us to maintain the documentation as the row number and all column names are defined in R/datasets_documentation.R. If these tests fail, update the relevant documentation, and then (ONLY THEN!) update the test expectations to match the new documentation.
+
+The fetch_ family of functions in R/fetch.R act as quick helpers that pull from the data sets we export, so users can ees-ily grab say a list of all Scottish Parliamentary Constituencies for 2024, rather than needing to pull in a whole data frame and process it.
+
+Often if adding a new year of data in, you will need to edit the year variables set near the start of the data-raw/ file and then also in the relevant fetch_ function @param year, as well as updating the public documentation of the data set in R/datasets_documentation.R.
+
```

Comment:
I sometimes get an error on `usethis::check_is_package()` deciding that I'm not in an r package when I try to run the geography data update scripts. I find it goes away if I restart the R session, but maybe worth adding a note on this.

Reply:
I haven't had this / can't work out how to recreate - could you suggest some text for me to add in? What does the error look like? @rmbielby

Reply:
![Screenshot 2024-09-16 103852](https://github.com/user-attachments/assets/a93ed21a-eb29-41ee-8337-1cd61e8f600e)

Reply:
Think I saw it briefly last week as well and then this morning. Not sure what triggers it yet. Maybe it's a very specific situation I've got myself in and won't effect others...

Reply:
Very possible. Though I'll stick it in either way - can't hurt!

Reply:
Added that now